drivers: pinmux: stm32h7: fix incorrect USART6 pin alternate functions

lochej opened the pull request:
Fixed alternate function of PG9/PG14 for USART6 with respect to the datasheet (p98-p99 stm32h743zi.pdf mcu datasheet)
Must be AF7 instead of AF8.


Validated on STM32H743ZI:
Enables the USART6 to be used on D0/D1 pins of the Nucleo H743ZI board.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

erwango commented:
@lochej Thanks for this contribution.
One comment though (but you don't need to modify anything):
In general, the rule is that people add their copyright only for major contribution on the file, which is not exactly the case here, but this will be the case with your ethernet driver proposal ;-).

lochej commented:
> @lochej Thanks for this contribution.
> One comment though (but you don't need to modify anything):
> In general, the rule is that people add their copyright only for major contribution on the file, which is not exactly the case here, but this will be the case with your ethernet driver proposal ;-).

You are absolutely right that sounds fair. I did this small PR get more familiar with the git contribution flow :) It's OK if my name doesn't show up in this file ^^

FRASTM commented on drivers/pinmux/stm32/pinmux_stm32h7.h:
```diff
@@ -507,7 +508,7 @@
 	(STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_PULLUP)
```
why then leaving USART6_RTS and CTS wrongly on  AF8,  AF7 is the correct setting.

Idem for PG15 below

lochej replied:
You are absolutely right ! I didn't have a look at the other pins as I didn't test them. I have submitted a new commit with all of the USART6 AF corrections. :+1 Thanks for your feedback

lochej committed:
drivers: pinmux: stm32h7: fix incorrect PG9/PG14 usart6 AF

Fixes AF of PG9/PG14 for usart6. Must be AF7 instead of AF8

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
drivers: pinmux: corrected all USART6 alternate functions

Fixed AF for USART 6 on PC6/PC7 PG8/PG9/PG12/PG13/PG14/PG15
All RTS/CTS/RX/TX pins now have correct AF7 instead of wrong AF8.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

ioannisg commented:
@lochej one more hint from me; you could just amend (commit --amend) the first patch and remove the copyright adding, so you don't have to do it and undo it in the following commit.

lochej commented:
> @lochej one more hint from me; you could just amend (commit --amend) the first patch and remove the copyright adding, so you don't have to do it and undo it in the following commit.

Thank you for your advice @ioannisg ! I have done the changes based on your recommendations. I'm still in the process of learning git and I'm getting more comfortable every day :smile:
